### PR TITLE
[WIP] initial Jenkinsfile using docker

### PR DESCRIPTION
**Pull request by dagar**
http://ci.px4.io:8080/blue/organizations/jenkins/DroneCore/detail/pr-jenkins/1/pipeline

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
+pipeline {
+  agent any
```
Comment (JonasVautherin) [inline Jenkinsfile]: So we only have linux agents, right? No MacOS from here?

Comment (dagar) [reply Jenkinsfile]: OSX machines are labelled mac. https://github.com/mavlink/qgroundcontrol/blob/master/Jenkinsfile#L55

Docker slaves are labelled docker.

**Comment by dagar** [comment]
TODO
 - create and archive packages
 - Examples: install package and build

**Comment by dagar** [comment]
Merge and keep going with this?

**Comment by hamishwillee** [comment]
When this is done it would be good if we (you guys) could provide an overview of the CI system for the docs. At the moment we have a little material on testing https://docs.dronecore.io/v/develop/en/contributing/test.html and building in docker: https://docs.dronecore.io/v/develop/en/contributing/build.html#building-in-docker

**Pull request merged**